resource/resource.go

```diff
@@ -0,0 +1,14 @@
+package resource
+
+import (
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sync"
+)
+
+// needs to be accessible by both project controllers and mgmt controllers
+var (
+	CleanerFuncMap = struct {
+		sync.RWMutex
+		M map[string]schema.GroupVersionResource
+	}{M: make(map[string]schema.GroupVersionResource)}
+)
```

Comment:
We might as well go the extra step and properly protect this map from concurrency issues.
- Make the M lowercase so that it is not publicly exposed.
- add a put method that is wrapped in Lock()
- add a Get function that returns a copy of the map

Reply:
rather than a copy of the map, return a list of GVRs

Reply:
I think it would be more useful in the future to have the map to check for existence ( O(1) vs O(n) for a list)